### PR TITLE
Holy Water Rework

### PR DESCRIPTION
**Pull request by Fox-McCloud**
An alternative to: https://github.com/ParadiseSS13/Paradise/pull/1937

Reworks how holy water decults people:

- Holy water no longer has a flat 10% chance to decult people every cycle its in them.
 - Instead, it's based on how long its in them: at 30 cycles (12 units metabolized--~60 seconds) it has a 5% chance each cycle to make cultists chant some cult related phrase.
 - At 75 cycles (30 units metabolized--~150 seconds) it has a 33% each cycle to decult

Should still allow holy water to deconvert people, but should be a bit less obvious and take a lot lot longer time for it to do so.

Related:
- As there's no longer an 80% chance to increase data (it's guaranteed now), the switch statement conditions for vampire's holy water effects have been increased by ~20%, rounded up.

### CODE REVIEW

**Comment by Robertha89** [comment]
I would like to see the metabolizing rate of Holy Water possible be increased.

**Comment by Earthdivine** [comment]
>it has a 5% chance each cycle to make cultists chant some cult related phrase.

Doesn't this make it even MORE of a valid hunting solution? I see no reason to not shoot syringes into people, as if they start screaming stuff randomly people will hear it, and if they run then that will get them arrested, not to mention it won't stop grenades from doing exactly what they do now, expose and possibly deconvert cultists.

**Comment by Fox-McCloud** [comment]
As it stands now, you can deconvert with a single drop of holy water if you're super lucky (yes, even 0.00001 units can deconvert, currently), which gives a visual message that it happened (much like rev). With this, they'll have to be hit with something and 60 seconds will have to pass before it even rolls for a chance for them to potentially start saying something--then a minute and a half before they start rolling for a chance to deconvert.

You're never going to be able to fully stop people from using holy water in grenades or syringe guns unless you make the stuff so horrendously lethal and toxic that it becomes potent enough for people to want to weaponize it for use against anyone.

**Comment by Aurorablade** [comment]
" it has a 5% chance each cycle to make cultists chant some cult related phrase."

....All i can think of is the Exorcist....

**Comment by Regen1** [comment]
A much better fix that what I managed to muster up, thanks for a better and more fleshed out method of dealing with it :+1:

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 1]
```diff
@@ -497,7 +507,8 @@ datum
 								M.IgniteMob()			//Only problem with igniting people is currently the commonly availible fire suits make you immune to being on fire
 								M.adjustFireLoss(3)		//Hence the other damages... ain't I a bastard?
 								M.mind.vampire.nullified = max(5, M.mind.vampire.nullified + 2)
-				holder.remove_reagent(src.id, 10 * REAGENTS_METABOLISM) //high metabolism to prevent extended uncult rolls.
+				holder.remove_reagent(src.id, 0.4)	//fixed consumption to prevent balancing going out of whack
```
Comment (TheDZD) [inline code/modules/reagents/Chemistry-Reagents.dm]: Couldn't you just set the metabolization_rate to 0.4 and avoid needing this here?

Comment (Fox-McCloud) [inline code/modules/reagents/Chemistry-Reagents.dm]: In this case, I'm not completely certain--perhaps. It means calling parent, but I'm not sure if that matters for us; I believe on some codebases, holy water is a child of water, so it makes sense NOT to call it.

**Comment by Fox-McCloud** [comment]
since our holy water isn't a child of water, I replaced just called parent.

**Pull request merged**